File: backend/Models/Entities/Student.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Models.Entities;

public class Student
{
    public int Id { get; set; }

    [Required]
    [MaxLength(150)]
    public string FullName { get; set; } = string.Empty;

    [Required]
    [MaxLength(255)]
    [EmailAddress]
    public string Email { get; set; } = string.Empty;

    [MaxLength(30)]
    public string? Phone { get; set; }

    [MaxLength(200)]
    public string? University { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<Application> Applications { get; set; } = new List<Application>();
}

File: frontend/src/app/models/student.ts
/**
 * Matches StudentDto returned by the API.
 * Navigation properties are no longer serialised, so there is no
 * applications array here.
 */
export interface Student {
  id: number;
  fullName: string;
  email: string;
  phone: string | null;
  university: string | null;
  /** ISO 8601 date-time string. */
  createdAt: string;
}

/** Matches StudentSummary, nested inside an application response. */
export interface StudentSummary {
  id: number;
  fullName: string;
  email: string;
}

/** Shape accepted by POST /api/Students and PUT /api/Students/{id}. */
export interface StudentRequest {
  fullName: string;
  email: string;
  phone?: string | null;
  university?: string | null;
}
